From 8a7973f4a1ab15b162e1df8e784518c9e4d584c6 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Sat, 9 Dec 2017 12:46:33 +0000
Subject: [PATCH 2/2] pppd: clear resolv.conf on link down

When the link goes down and we're not doing dial on demand, the name
servers are no longer valid, so remove them from resolv.conf.

Signed-off-by: Mans Rullgard <mans@mansr.com>
Upstream-Status: Inappropriate [Venus specific]
---
 pppd/ipcp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pppd/ipcp.c b/pppd/ipcp.c
index 4be2d7deb2b7..27f2093a8c04 100644
--- a/pppd/ipcp.c
+++ b/pppd/ipcp.c
@@ -2031,6 +2031,8 @@ ipcp_down(f)
 	sifdown(f->unit);
 	ipcp_clear_addrs(f->unit, ipcp_gotoptions[f->unit].ouraddr,
 			 ipcp_hisoptions[f->unit].hisaddr, 0);
+	if (usepeerdns)
+	    create_resolv(0, 0);
     }
 
     /* Execute the ip-down script */
-- 
2.26.2

